rules/library.bzl

```diff
@@ -841,62 +841,47 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
-    # Other dependencies' headermaps will be generated by li_ios_framework
-    # rules.
-    headermap(
-        name = public_hmap_name,
-        namespace = namespace,
-        hdrs = [public_hdrs_filegroup],
-        tags = _MANUAL,
-    )
-    private_dep_names.append(public_hmap_name)
+        # Public hmaps are for vendored static libs to export their header only.
+        # Other dependencies' headermaps will be generated by li_ios_framework
+        # rules.
+        headermap(
+            name = public_hmap_name,
+            namespace = namespace,
+            hdrs = [public_hdrs_filegroup],
+            tags = _MANUAL,
+        )
+        private_dep_names.append(public_hmap_name)
+        _append_headermap_copts(public_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
 
-    private_hmap_name = name + "_private_hmap"
-    private_angled_hmap_name = name + "_private_angled_hmap"
-    private_hdrs_filegroup = name + "_private_hdrs"
-    private_angled_hdrs_filegroup = name + "_private_angled_hdrs"
+    if len(objc_non_exported_hdrs + objc_private_hdrs) > 0:
+        private_hmap_name = name + "_private_hmap"
+        private_hdrs_filegroup = name + "_private_hdrs"
 
-    native.filegroup(
-        name = private_hdrs_filegroup,
-        srcs = objc_non_exported_hdrs + objc_private_hdrs + objc_hdrs,
-        tags = _MANUAL,
-    )
-    native.filegroup(
-        name = private_angled_hdrs_filegroup,
-        srcs = objc_non_exported_hdrs + objc_private_hdrs,
-        tags = _MANUAL,
-    )
+        native.filegroup(
+            name = private_hdrs_filegroup,
+            srcs = objc_non_exported_hdrs + objc_private_hdrs,
+            tags = _MANUAL,
+        )
 
-    headermap(
-        name = private_hmap_name,
-        hdrs = [private_hdrs_filegroup],
-        tags = _MANUAL,
-    )
-    private_dep_names.append(private_hmap_name)
-    headermap(
-        name = private_angled_hmap_name,
-        namespace = namespace,
-        hdrs = [private_angled_hdrs_filegroup],
-        tags = _MANUAL,
-    )
-    private_dep_names.append(private_angled_hmap_name)
+        headermap(
+            name = private_hmap_name,
+            hdrs = [private_hdrs_filegroup],
+            tags = _MANUAL,
+        )
+        private_dep_names.append(private_hmap_name)
+        _append_headermap_copts(private_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
 
     ## END HMAP
 
-    _append_headermap_copts(private_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
-    _append_headermap_copts(public_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
-    _append_headermap_copts(private_angled_hmap_name, "-I", additional_objc_copts, additional_swift_copts, additional_cc_copts)
-    _append_headermap_copts(private_hmap_name, "-iquote", additional_objc_copts, additional_swift_copts, additional_cc_copts)
```

Comment:
Sorry noticed something I should've asked in my previous review.

Since this is being deleted mind clarifying how it maps to the final code in this PR? E.g. why `private_angled_hmap_name` and the `"-iquote"` line are not being used anymore?

Reply:
These are moved to the `if`'s above.